### ISSUE TITLE
Training an additional entity for the german core model failed

### ISSUE THREAD

**Issue by fteischi** [explosion/spaCy#3279]
Hello,

I tried to add an entity to the german core model of spacy ('de_core_news_sm'), since I have come to understand that the pre-trained german model does not include an entity for money - such as the english model.

For some reason, I am not able to add an entity to the german model. The code from below -despite following the instruction on the website https://spacy.io/usage/training -wouldn't recognize entities related to money (i.e. Franken, Euro) from the training data (TRAIN_DATA).

I was wondering, where the problem lies.

Many thanks and best regards.

## How to reproduce the behaviour
import os, glob
import PyPDF2

import pandas as pd

import plac
import random
from pathlib import Path

import spacy #NLP library
from spacy import displacy
from spacy.util import minibatch, compounding

from tqdm import tqdm

LABEL = 'GELD'

TRAIN_DATA = [
    ("Ein Konzerngewinn von 1053 Millionen Euro.", {"entities": [(38, 41, "GELD")]}),
    ("Der Geschäftsaufwand sank um 430 Millionen Franken.", {"entities": [(44, 50, "GELD")]}),
    ("Der Vorsteuergewinn stieg um 1178 Millionen Franken.", {"entities": [(45, 51, "GELD")]}),
    ("Der Gewinn beträgt 1 Million Franken.", {"entities": [(30, 36, "GELD")]}),
    ("Wieviel sind 2 Millionen Franken in Euro.", {"entities": [(27, 33, "GELD"),(38, 41, "GELD")]}),,
]

@plac.annotations(
    model=("Model name. Defaults to blank 'en' model.", "option", "m", str),
    new_model_name=("New model name for model meta.", "option", "nm", str),
    output_dir=("Optional output directory", "option", "o", Path),
    n_iter=("Number of training iterations", "option", "n", int))

def main(model=None, new_model_name='animal', output_dir=None, n_iter=15):
    """Set up the pipeline and entity recognizer, and train the new entity."""
    if model is not None:
        nlp = spacy.load(model)  # load existing spaCy model
        print("Loaded model '%s'" % model)
    else:
        nlp = spacy.blank('en')  # create blank Language class
        print("Created blank 'en' model")
    # Add entity recognizer to model if it's not in the pipeline
    # nlp.create_pipe works for built-ins that are registered with spaCy
    if 'ner' not in nlp.pipe_names:
        ner = nlp.create_pipe('ner')
        nlp.add_pipe(ner)
    # otherwise, get it, so we can add labels to it
    else:
        ner = nlp.get_pipe('ner')
    ner.add_label(LABEL)   # add new entity label to entity recognizer
    if model is None:
        optimizer = nlp.begin_training()
    else:
        # Note that 'begin_training' initializes the models, so it'll zero out
        # existing entity types.
        optimizer = nlp.entity.create_optimizer()

    # get names of other pipes to disable them during training
    other_pipes = [pipe for pipe in nlp.pipe_names if pipe != 'ner']
    with nlp.disable_pipes(*other_pipes):  # only train NER
        for itn in tqdm(range(n_iter)):
            random.shuffle(TRAIN_DATA)
            losses = {}
            # batch up the examples using spaCy's minibatch
            batches = minibatch(TRAIN_DATA, size=compounding(4., 32., 1.0001))
            #batches = get_batches(TRAIN_DATA, 'ner')
            for batch in batches:
                texts, annotations = zip(*batch)
                nlp.update(texts, annotations, sgd=optimizer, drop=0.35,
                           losses=losses)
            #print('Losses', losses)

    # test the trained model
    test_text = 'Wieviel Euro sind 2 Millionen Franken für Shaka Khan?'
    doc = nlp(test_text)
    print("Entities in '%s'" % test_text)
    for ent in doc.ents:
        print(ent.label_, ent.text)

    # save model to output directory
    if output_dir is not None:
        output_dir = Path(output_dir)
        if not output_dir.exists():
            output_dir.mkdir()
        nlp.meta['name'] = new_model_name  # rename model
        nlp.to_disk(output_dir)
        print("Saved model to", output_dir)

        # test the saved model
        print("Loading from", output_dir)
        nlp2 = spacy.load(output_dir)
        doc2 = nlp2(test_text)
        for ent in doc2.ents:
            print(ent.label_, ent.text)

main(model='de_core_news_sm', output_dir='german_trai_ner')

## Output
Loaded model 'de_core_news_sm'
100%|██████████| 15/15 [00:10<00:00,  1.46it/s]
Entities in 'Wieviel Euro sind 2 Millionen Franken für Shaka Khan?'
Saved model to german_trai_ner
Loading from german_trai_ner

## Your Environment
* Operating System: Windows 7
* Python Version Used: 3.6.8.
* spaCy Version Used: 2.0.18
* Environment Information: Python runs within Anaconda

**Comment by honnibal** [comment]
Hi,

I fixed this example recently for `spacy-nightly`. You should soon be able to get the v2.1.0a9 version, which should work if you run it with the version of `examples/training/train_new_entity_type.py` that's on the develop branch.

**Comment by honnibal** [comment]
v2.1.0a9 now uploaded. Please reopen if it doesn't work with the example on the develop branch.
